src/resolve/resolved-mdns.c

```diff
@@ -397,6 +406,20 @@ static int on_mdns_packet(sd_event_source *s, int fd, uint32_t revents, void *us
                                 log_debug("Got a goodbye packet");
                                 /* See the section 10.1 of RFC6762 */
                                 rr->ttl = 1;
+
+                                /* Look at the cache 1 second later and remove stale entries.
+                                 * This is particularly useful to keep service browsers updated on service removal,
+                                 * as there are no other reliable triggers to propogate that info. */
+                                r = sd_event_add_time_relative(
+                                                m->event,
+                                                NULL,
+                                                CLOCK_BOOTTIME,
+                                                USEC_PER_SEC,
+                                                0,
+                                                mdns_goodbye_callback,
+                                                scope);
```

Comment:
this will result in memory corruption: let's say a scope goes away (because a network interface goes away), but the event source is still queued. it will be called pointing to a scope that doesn't exist anymore. Bad things will then happen.

if you embedd a pointer to the eent source in DnsScope object, then you can also delete the event source whenever the scope goes away.